### PR TITLE
Alter Posthook to use a Queue and add in API Method to run queue jobs

### PR DESCRIPTION
**Pull request by seamuslee001**
This should give us much more control over when the placement operations happen as well but would also mean that transactions shouldn't fail hopefully

### CODE REVIEW

File: CRM/CiviGeometry/Helper.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
+<?php
+
+/**
+ * This is a helper class for the queue functionality.
+ * It is a singleton class because it will hold the queue object for our extension
+ */
+class CRM_CiviGeometry_Helper {
+  const QUEUE_NAME = 'au.org.greens.civigeomery.queue';
```
Comment (johntwyman) [inline CRM/CiviGeometry/Helper.php]: Typo here.

File: CRM/CiviGeometry/Tasks.php [hunk 2]
```diff
@@ -6,19 +6,28 @@ class CRM_CiviGeometry_Tasks {
...
-      'geometry_a' => 0,
-      'geometry_b' => 'POINT(' . $address['geo_code_2'] . ' ' . $address['geo_code_1'] . ')',
-    ])['values'];
-    if (!empty($geometry_ids)) {
-      foreach ($geometry_ids as $geometry_id) {
-        civicrm_api3('Address', 'creategeometries', [
-          'address_id' => $id,
-          'geometry_id' => $geometry_id,
-        ]);
+    try {
```
Comment (johntwyman) [inline CRM/CiviGeometry/Tasks.php]: +1 for try/catch. Nice.

**Comment by andrew-cormick-dockery** [comment]
OK I see what you're doing here.  How are the queues run?  Do we need a new cronjob?

**Comment by andrew-cormick-dockery** [comment]
> 
> 
> OK I see what you're doing here. How are the queues run? Do we need a new cronjob?

Sorry, just saw the other PR

**Comment by seamuslee001** [comment]
I modeled my code on here https://github.com/jaapjansma/nl.edeveloper.queuehowto and as you can see in the changes to the tests that the queue is being processed successfully and then causing correct things to happen

**Pull request merged**